tests/conftest.py

```diff
@@ -577,6 +590,41 @@ def pytest_fixture_setup(fixturedef, request):
             for mark in request.node.iter_markers():
                 if mark.name == "config_index":
                     ocsci_config.switch_ctx(mark.args[0])
+    _switch_context_helper(request)
+
+
+def _switch_context_helper(request):
```

Comment:
why is it defined as a private function? Can't it be used outside this file?

Reply:
code was extracted from the pytest_fixture_setup to reuse in finalizers. 
This function was to use it in module only in fixture functions and helpers where request arg is visible. Can't see now where it can be reused outside of the module.
But this can change in future, we can set it public when use case will appear.

Reply:
we have moved from this draft to https://github.com/red-hat-storage/ocs-ci/pull/12041, it is updated and verified